### PR TITLE
Fixed an issue causing failures when using change feed in batch mode with a batch location and `ChangeFeedBatch.planInputPartitions` is called multiple times

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Fixed an issue causing failures when using change feed in batch mode with a batch location and `ChangeFeedBatch.planInputPartitions` is called multiple times - (for example because physcial query plan gets retrieved) and some changes have been made in the monitoied container between those calls.)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala [hunk 1]
```diff
@@ -154,15 +154,17 @@ private class ChangeFeedBatch
...
-
           if (existingLatestOffset != latestOffsetJson) {
-            log.logError(msg)
+            val msg = s"Cannot update latest offset at location '$latestOffsetLocation' for batchId: $batchId " +
+              s"-> existing latestOffset: '$existingLatestOffset' failed to persist " +
+              s"new latestOffset: '$latestOffsetJson'- will continue with existing latest offset."
+            log.logWarning(msg)
 
-            throw new IllegalStateException(msg)
+            latestOffset = ChangeFeedOffset.fromJson(existingLatestOffset)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala:163]: should we also validate the collectionRid is same?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala:163]: Not sure it is really needed - but won't hurt. Added it.

**Review by tvaron3 — APPROVED** [review]
LGTM

**Pull request merged**